cni/network/invoker_cns.go

```diff
@@ -135,7 +135,7 @@ func (invoker *CNSIPAMInvoker) Add(addConfig IPAMAddConfig) (IPAMAddResult, erro
 
 		ncgw := net.ParseIP(info.ncGatewayIPAddress)
 		if ncgw == nil {
-			if (invoker.ipamMode != util.V4Overlay) && (invoker.ipamMode != util.DualStackOverlay) {
+			if (invoker.ipamMode != util.V4Overlay) && (invoker.ipamMode != util.DualStackOverlay) && (invoker.ipamMode != util.Overlay) {
```

Comment:
we probably need to also take care of those files with consolidated flag too:

(1) network_windows.go under network folder:
Line 431:
if nwInfo.IPV6Mode == string(util.DualStackOverlay) => we assign this ipamMode to IPV6Mode to add ipv6 net routes to windows node

Only it's dualstack mode, then we add net routes to windows node; if it's v4 overlay, then we do not add these ipv6 net rules.

(2) network.go under cni/network folder:
Line 777:
// inject ipv6 routes to Linux pod
epInfo.IPV6Mode = string(util.IpamMode(opt.nwCfg.IPAM.Mode))
if IPAM mode becomes overlay, also need to check if it's dualstack mode to inject ipv6 routes to Linux pod.

Reply:
Ah yes, these two cases mean that there is a difference between v4overlay and dualstackoverlay ipam modes @ashvindeodhar 

So I think we need to keep them separate

Shall I abandon this draft or change "v4overlay" to "overlay" ? I don't think we can consolidate "v4overlay" and "dualstackoverlay" to just "overlay", there needs to be a separation

Reply:
yes, we probably can use "executionMode" in the conflist file to differentiate it if we use consolidated "overlay" flag, need design this

Reply:
I will check with HNS team and Neha if these routes (for windows) are needed also in v4 then follow up here

Reply:
Yes, HNS said we want the 1 ipv4 route for both dualstack and v4overlay, which we can do eventually but not in this PR

Reply:
Removing 1 redundant route in this PR:
https://github.com/Azure/azure-container-networking/pull/2002/files